lib/pure/includes/osenv.nim

```diff
@@ -45,8 +45,8 @@ else:
 
   proc c_getenv(env: cstring): cstring {.
     importc: "getenv", header: "<stdlib.h>".}
-  proc c_putenv(env: cstring): cint {.
-    importc: "putenv", header: "<stdlib.h>".}
+  proc c_setenv(envname: cstring, envval: cstring, overwrite: cint): cint {.
+    importc: "setenv", header: "<stdlib.h>".}
   proc c_unsetenv(env: cstring): cint {.
     importc: "unsetenv", header: "<stdlib.h>".}
```

Comment:
add test in tests/stdlib/tos.nim (in block that tests putEnv)
the test should contain a test that fails before your PR and works after your PR (see https://github.com/nim-lang/Nim/issues/18502)